### DISCUSSION TITLE
What is the ormar way to create an entity with relations via Fastapi?

### DISCUSSION THREAD

**Discussion by Strernd** [collerek/ormar#617] [Q&A]
I have some models with different many to many relationships. For the sake of simplicity assume we have the two models `Chocolate` and `Assortment` a `Chocolate` can be in different `Assortments` and a `Assortment` contains multiple different `Chocolate` entities. 
 
I've created endpoints with Fastapi for `chocolate` and created some entities and have their ids. 
Now I want to create an `Assortment` by passing 

```json
{
  "name": "Finest Selection",
  "price": 10,
  "chocolates": ["abcd-1234-chocid-01", "abcd-1234-chocid-02"]
}
```

to the `POST /assortment` endpoint. Of course I want to add the `Chocolate` entities to the `Assortment` entity by only passing the ids. 

I haven't found a way for this that seems to be intended. 
Do I have to break down the pydantic model (so that it ignores the ids) and then call `assortment.chocolates.add(...)` for every `Chocolate` entity that I need to fetch from the database?

**Comment by collerek** [comment]
If you want to save the whole tree in one call you can call `Model.save_related(follow=True, save_all=True)`. It will upsert all models including nested children, with through models for m2m relations too.

As of now `add` only accepts the model instance but will look into it.
